test/ruby/test_io.rb

```diff
@@ -1898,6 +1898,116 @@ def test_set_lineno_gets
     end)
   end
 
+  def test_readline_bad_param_raises
+    File.open(__FILE__) do |f|
+      assert_raise(TypeError) do
+        f.readline Object.new
+      end
+    end
+
+    File.open(__FILE__) do |f|
+      assert_raise(TypeError) do
+        f.readline 1, 2
+      end
+    end
+  end
+
+  def test_readline_raises
+    File.open(__FILE__) do |f|
+      File.open(__FILE__) do |f2|
+        assert_equal f2.read, f.readline(nil)
```

Comment:
Why not just `assert_equal File.read(__FILE__), f.readline(nil)`?

Reply:
Fixed, thanks!